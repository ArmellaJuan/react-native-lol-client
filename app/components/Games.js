import React, { Component, PropTypes } from 'react';
import { StyleSheet, ActivityIndicator, ScrollView, TouchableHighlight, View, Text, Image, ListView } from 'react-native';

import RowInfo from './RowInfo';
import Kda from './Kda';

import Icon from 'react-native-vector-icons/Entypo';

import GlobalStyles from '../util/globalStyles';

export default class Games extends Component {
  constructor(props) {
    super(props);
  }


  componentDidMount(){
    this.props.onRequestRecentGames(this.props.summonerId);
  }

  componentWillUnmount(){
    this.props.onExit();
  }

  renderList(){

    if(!this.props.loading){
      let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => JSON.stringify(r1) !== JSON.stringify(r2) });
      let dataSource = ds.cloneWithRows(this.props.games);
      return (<ListView
                style= { styles.matchs }
                dataSource={dataSource}
                renderRow={this.renderRow.bind(this) }
              />);
    }

  }

  renderRow(gameData) {    
    return (
      <TouchableHighlight style={ [ styles.rowContainer, GlobalStyles.shadow ] } onPress={ ()=>this.props.onGameSelected( gameData.index ) } underlayColor='dodgerblue' >
        
        <View style={ [styles.row, gameData.victory? styles.victory : styles.defeat] } >
        
            <Image style={ styles.image } source= { gameData.champion? { uri: gameData.champion.imageUrl } : null } />
            
            <View style= { styles.matchStatFirstColumn } >
              <Text style={ styles.matchType } >{gameData.type}</Text>
              <Text style={ styles.timeLabel } >{gameData.date}</Text>
              <Text style={ styles.timeLabel }>{gameData.timePlayed}</Text>
            </View>
            
            <View style= { styles.matchStatsSecondColumn }>
              <RowInfo valueStyle={ [gameData.victory? styles.victoryLabel : styles.defeatLabel] } fontSize={10} label='Result' value={gameData.victory? 'Victory' : 'Defeat'} /> 
              <View style={ styles.infoRow } >
                <Text style={[ styles.infoLabel, { flex: 1 } ]} >KDA</Text>
                <Kda kills={gameData.kills} deaths={gameData.deaths} assists={gameData.assists} />
              </View>
            </View>

            <View  style= { styles.toDetail }>
              <Icon name="chevron-thin-right" size={25} color="dodgerblue" />
            </View>

      </View>

    </TouchableHighlight>
    );
  } 

  render() {
    return (
        <View style={styles.background}>
           <ScrollView >
             {this.props.loading ? (
                  <View>
                    <ActivityIndicator
                    style={[styles.loading]}
                    />
                  </View>
                  ) : (
                     <View style= { styles.scrollViewContainer } >
                      { this.renderList() }     
                    </View>
                )}
          </ScrollView>
        </View>
    );
  }
}


Games.propTypes = {
  summonerId: PropTypes.number.isRequired,
  loading: PropTypes.bool.isRequired,
  onRequestRecentGames: PropTypes.func.isRequired,
  games: PropTypes.array,
  onExit: PropTypes.func,
  onGameSelected: PropTypes.func.isRequired
};


const styles = StyleSheet.create({
  scrollViewContainer:{
    padding: 20,
  },
  scrollView:{
    padding: 10 
  },
  background: {
    flex: 1,
    backgroundColor: 'aliceblue',
  },
  infoRow:{
    flexDirection: 'row',
    flex: 1
  },
  infoLabel: {
    flex:1,
    fontWeight: 'bold',
    fontSize: 10
  },
  rowContainer:{
    margin: 5,
    backgroundColor: 'white'
  },
  row: {
    height: 80,
    flexDirection: 'row',
    padding: 10,
    flex: 1,
    alignItems: 'center',
  },
  victory: {
    backgroundColor: '#d4e7f7'
  },
  defeat: {
    backgroundColor: '#f1dcda'
  },
  loading: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
    height: 80
  },
  image: {
    width: 50, 
    height: 50,
    borderRadius: 25
  },
  matchType: {
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 10,
    marginBottom: 2
  },
  timeLabel:{
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 10,
  },
  victoryLabel: {
    color: '#1a78ae',
    fontWeight: 'bold',
  },
  defeatLabel:{
    color: '#c6443e',
    fontWeight: 'bold',
  },
  matchStatFirstColumn:{
    paddingRight: 10, 
    paddingLeft: 10, 
    flex: 1.5,  
    alignItems: 'center', 
    justifyContent : 'center',
  },
  matchStatsSecondColumn: {
    flex: 1.5, 
    alignItems: 'center',  
  },
  toDetail:{
    flex: 0.5 
  }
});
