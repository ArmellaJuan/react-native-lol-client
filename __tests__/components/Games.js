import 'react-native';
import React from 'react';
import Games from '../../app/components/Games';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';



test('renders correctly', () => {

  let games = [
    {  
      "id":423574538,
      "type":"Normal",
      "victory":false,
      "kills":5,
      "assists":2,
      "deaths":13,
      "date":"8 hours ago",
      "timePlayed":"32m 41s",
      "index":0,
      "totalDamageDealt":133607,
      "totalDamageDealtToChampions":13888,
      "physicalDamageDealtToChampions":12585,
      "magicDamageDealtToChampions":1302,
      "fellowPlayers":[  
        {  
          "summonerId":375902,
          "teamId":100,
          "championId":120
        },
        {  
          "summonerId":391041,
          "teamId":100,
          "championId":16
        }
      ]
    },
    {  
      "id":423538758,
      "type":"Assassinate",
      "victory":true,
      "kills":4,
      "assists":7,
      "deaths":6,
      "date":"9 hours ago",
      "timePlayed":"13m 32s",
      "index":1,
      "totalDamageDealt":28777,
      "totalDamageDealtToChampions":11710,
      "physicalDamageDealtToChampions":1329,
      "magicDamageDealtToChampions":9839,
      "fellowPlayers":[  
        {  
          "summonerId":1813407,
          "teamId":200,
          "championId":121
        },
        {  
          "summonerId":111665,
          "teamId":100,
          "championId":121
        }
      ]
    }];


  const tree = renderer.create(
    <Games games={games} summonerId={1} loading={false} onRequestRecentGames={ () => {} } 
      onExit= {() => {} } onGameSelected={ () => {} }
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});