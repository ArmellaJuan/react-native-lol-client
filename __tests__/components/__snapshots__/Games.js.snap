exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "aliceblue",
      "flex": 1,
    }
  }>
  <ScrollView>
    <View
      style={
        Object {
          "padding": 20,
        }
      }>
      <ScrollView
        dataSource={
          ListViewDataSource {
            "items": 2,
          }
        }
        initialListSize={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={1000}
        onKeyboardDidHide={undefined}
        onKeyboardDidShow={undefined}
        onKeyboardWillHide={undefined}
        onKeyboardWillShow={undefined}
        onLayout={[Function]}
        onScroll={[Function]}
        pageSize={1}
        removeClippedSubviews={true}
        renderRow={[Function]}
        scrollEventThrottle={50}
        scrollRenderAheadDistance={1000}
        stickyHeaderIndices={Array []}
        style={undefined}>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "white",
                "elevation": 3,
                "margin": 5,
              },
            ]
          }
          testID={undefined}>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 80,
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#f1dcda",
                },
              ]
            }>
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": null,
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "flex": 1,
                  "height": 50,
                  "width": 50,
                }
              } />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1.5,
                  "justifyContent": "center",
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginBottom": 2,
                    "textAlign": "center",
                  }
                }>
                Normal
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 11,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }>
                8 hours ago
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 11,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }>
                32m 41s
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1.5,
                }
              }>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 35,
                    "justifyContent": "center",
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontSize": 12,
                      },
                      Object {},
                    ]
                  }>
                  Result
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      Object {
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#c6443e",
                          "fontWeight": "bold",
                        },
                      ],
                    ]
                  }>
                  Defeat
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "fontSize": 12,
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 1,
                      },
                    ]
                  }>
                  KDA
                </Text>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#1a78ae",
                          "fontWeight": "bold",
                        },
                      ]
                    }>
                    5
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 12,
                        "fontWeight": "bold",
                      }
                    }>
                    /
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#c6443e",
                          "fontWeight": "bold",
                        },
                      ]
                    }>
                    13
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 12,
                        "fontWeight": "bold",
                      }
                    }>
                    /
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#1a78ae",
                          "fontWeight": "bold",
                        },
                      ]
                    }>
                    2
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0.5,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "dodgerblue",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "white",
                "elevation": 3,
                "margin": 5,
              },
            ]
          }
          testID={undefined}>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 80,
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#d4e7f7",
                },
              ]
            }>
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": null,
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "flex": 1,
                  "height": 50,
                  "width": 50,
                }
              } />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1.5,
                  "justifyContent": "center",
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginBottom": 2,
                    "textAlign": "center",
                  }
                }>
                Assassinate
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 11,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }>
                9 hours ago
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 11,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }>
                13m 32s
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1.5,
                }
              }>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 35,
                    "justifyContent": "center",
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontSize": 12,
                      },
                      Object {},
                    ]
                  }>
                  Result
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      Object {
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#1a78ae",
                          "fontWeight": "bold",
                        },
                      ],
                    ]
                  }>
                  Victory
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "fontSize": 12,
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 1,
                      },
                    ]
                  }>
                  KDA
                </Text>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#1a78ae",
                          "fontWeight": "bold",
                        },
                      ]
                    }>
                    4
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 12,
                        "fontWeight": "bold",
                      }
                    }>
                    /
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#c6443e",
                          "fontWeight": "bold",
                        },
                      ]
                    }>
                    6
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 12,
                        "fontWeight": "bold",
                      }
                    }>
                    /
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#1a78ae",
                          "fontWeight": "bold",
                        },
                      ]
                    }>
                    7
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0.5,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "dodgerblue",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                
              </Text>
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  </ScrollView>
</View>
`;
